src/datasources/targeted-messaging/targeted-messaging.datasource.ts

```diff
@@ -89,7 +121,7 @@ export class TargetedMessagingDatasource
     return {
       id: targetedSafe.id,
       address: targetedSafe.address,
-      outreachId: targetedSafe.outreach_id,
```

Comment:
Nit: I know it's not possible to change, but it feels more "correct" to return that from the database/cache.

Reply:
I think that would require to refactor `CachedQueryResolver` because it is currently typed. And the return type `TargetedSafe` does not contain an `outreach_id` but a `outreachId`, which does not exist at the database level. I agree this is a bit annoying right now, but hopefully we can reevaluate the types after the ORM migration. I feel some of these _problems_ will be vanished. 

And as you said, `args.outreachId` has to be correct and has to exist at the database level, because it is involved in the db query. So there's no room for inconsistencies IMO. I'd wait until the ORM migration is done to tackle this kind of detail, wdyt?

Reply:
> the return type TargetedSafe does not contain an outreach_id but an outreachId, which does not exist at the database level

This means the domain entity is being used for the database though, which is inherently incorrect. Can we create a second entity for that coming from the database?

Reply:
I've included database entities in [7a04f17](https://github.com/safe-global/safe-client-gateway/pull/2000/commits/7a04f17e5d1d62a8499e268baeaf2f894a5e2717). I think this introduces a patter we could use in other places, but as I said I think this approach might also be deprecated soon with the introduction of an ORM. 

If, for some reason, the pattern is not deprecated by the ORM, I'd like to establish it a bit more in a separate PR by adding a mapper to the database entities, so it would handle the database<>domain conversion in a centralised place.

This also solves the internal `string/Date` duality in the tests, as the entity being sent to the repository is mapped inside the datasources layer.